### PR TITLE
fix recommendation Bug

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
add code for handle enum case in CodeGen

### CODE REVIEW

**Comment by LittleLittleCloud** [comment]
#4579

File: test/Microsoft.ML.CodeGenerator.Tests/TrainerGeneratorTests.cs [hunk 1]
```diff
@@ -295,14 +296,21 @@ public void MatrixFactorizationAdvancedTest()
                 {"MatrixColumnIndexColumnName","userId" },
                 {"MatrixRowIndexColumnName","movieId" },
                 {"LabelColumnName","rating" },
+                {nameof(MatrixFactorizationTrainer.Options.NumberOfIterations), 10 },
+                {nameof(MatrixFactorizationTrainer.Options.LearningRate), 0.01f },
+                {nameof(MatrixFactorizationTrainer.Options.ApproximationRank), 8 },
+                {nameof(MatrixFactorizationTrainer.Options.Lambda), 0.01f },
+                {nameof(MatrixFactorizationTrainer.Options.LossFunction), MatrixFactorizationTrainer.LossFunctionType.SquareLossRegression },
+                {nameof(MatrixFactorizationTrainer.Options.Alpha), 1f },
+                {nameof(MatrixFactorizationTrainer.Options.C), 0.00001f },
```
Comment (justinormont) [inline test/Microsoft.ML.CodeGenerator.Tests/TrainerGeneratorTests.cs]: Thanks for adding the MF options to a test.

File: src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/CodeGenerator.cs [hunk 2]
```diff
@@ -329,18 +329,11 @@ internal IList<(string, string[])> GenerateTransformsAndUsings(IEnumerable<Pipel
...
-            }
-            catch (Exception)
-            {
-                return (string.Empty, new string[0]);
-            }
+            var node = _pipeline.Nodes.Where(t => t.NodeType == PipelineNodeType.Trainer).First();
+            ITrainerGenerator generator = TrainerGeneratorFactory.GetInstance(node);
+            var trainerString = generator.GenerateTrainer();
+            var trainerUsings = generator.GenerateUsings();
+            return (trainerString, trainerUsings);
```
Comment (justinormont) [inline src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/CodeGenerator.cs]: I'm not sure why the try/catch was needed. If no one else knows, we may want drill down in the code, or run a variety of datasets thru the CLI to show that nothing is expected to throw.

The old catch seem to eat the error without a hint of what went wrong.

Comment (LittleLittleCloud) [reply src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/CodeGenerator.cs]: Yeah, that's why I removed that part of code

**Pull request merged**